lisa/util/shell.py

```diff
@@ -44,25 +45,103 @@ def __init__(
             raise LisaException("username must be set")
 
 
+class WindowsShellType(object):
+    """
+    Windows command generator
+    Support get pid, set envs, and cwd
+    Doesn't support kill, it needs overwrite spur.SshShell
+    """
+
+    supports_which = False
+
+    def generate_run_command(
+        self,
+        command_args: List[str],
+        store_pid: bool = False,
+        cwd: Optional[str] = None,
+        update_env: Optional[Dict[str, str]] = None,
+        new_process_group: bool = False,
+    ) -> str:
+        commands = []
+
+        if store_pid:
+            commands.append(
+                'powershell "(gwmi win32_process|? processid -eq $pid).parentprocessid"'
+                " &&"
+            )
+
+        if cwd is not None:
+            commands.append(f"cd {cwd} 2>&1 && echo spur-cd: 0 ")
+            commands.append("|| echo spur-cd: 1 && exit 1 &")
+
+        if update_env:
+            update_env_commands = [
+                "set {0}={1}".format(key, value) for key, value in update_env.items()
```

Comment:
Do we need to sanitize these?

Reply:
couldn't find a clean way to do it in windows command. Let user handle that..